.github/actions/roundup_summary.py

```diff
@@ -0,0 +1,107 @@
+import json
+import os
+from datetime import date, timedelta
+from typing import Iterable
+
+import requests
+
+ISSUE_ENDPOINT = "https://github.com/python/cpython/issues"
+SEARCH_ENDPOINT = "https://api.github.com/search/issues"
+GRAPHQL_ENDPOINT = "https://api.github.com/graphql"
+MAILGUN_ENDPOINT = "https://api.mailgun.net/v3/mg.python.org"
+
+
+def get_issue_counts(token: str) -> tuple[int, int]:
+    """use the GraphQL API to get the number of opened and closed issues
+    without having to query every single issue and count them."""
+    data = {
+        "query": """
+        {
+            repository(owner: "python", name: "cpython") {
+                open: issues(states: OPEN) { totalCount }
+                closed: issues(states: CLOSED) { totalCount }
+            }
+        }
+        """
+    }
+    response = requests.post(GRAPHQL_ENDPOINT, json=data, headers={
+        "Authorization": f"Bearer {token}",
+        "accept": "application/vnd.github.v3+json"
+    })
+    repo = response.json()["data"]["repository"]
+    open = repo["open"]["totalCount"]
+    closed = repo["closed"]["totalCount"]
+    return open, closed
+
+def get_issues(filters: Iterable[str], token: str, all_: bool = True):
+    """return a list of results from the Github search API"""
+    # TODO: if there are more than 100 issues, we need to include pagination
```

Comment:
Should this TODO be addressed before merge? If not is there something we should put in place to make clear that there weren't magically just 100 issues a given week?